src/cascadia/UnitTests_SettingsModel/SerializationTests.cpp

```diff
@@ -1065,4 +1076,123 @@ namespace SettingsModelUnitTests
 
         VERIFY_ARE_EQUAL(sendInputCmd1.ID(), sendInputCmd1.ID());
     }
+
+    void SerializationTests::RoundtripLegacyToModernActions()
+    {
+        static constexpr std::string_view oldSettingsJson{ R"(
+        {
+            "actions": [
+                {
+                    "name": "foo",

```

Comment:
can you test what happens when you have two different legacy actions with the same _name_ but different commands?

Reply:
Added! Unfortunately could not use actions with no parameters since those will get deleted (since they exist in the defaults!)